logstash-core/lib/logstash/settings.rb

```diff
@@ -140,6 +137,11 @@ def validate_all
       end
     end
 
+    def ==(other)
+      return false unless other.kind_of?(::LogStash::Settings)
+      self.to_hash == other.to_hash
+    end
```

Comment:
+1, I've looked at `to_hash` method of individual setting. I believe it work for normal type, but shouldn't we change the behavior of `to_hash` in the case of `ArrayCoercible`?